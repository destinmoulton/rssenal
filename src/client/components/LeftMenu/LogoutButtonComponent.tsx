import * as React from "react";

import { Button, Popup } from "semantic-ui-react";

export interface ILogoutButtonComponentProps {
    logoutUser?: () => void;
}

class LogoutButtonComponent extends React.Component<
    ILogoutButtonComponentProps
> {
    constructor(props: ILogoutButtonComponentProps) {
        super(props);

        this._handleClickLogout = this._handleClickLogout.bind(this);
    }

    _handleClickLogout() {
        this.props.logoutUser();
    }

    render() {
        return (
            <Popup
                trigger={
                    <Button
                        className="rss-leftmenu-button-logout"
                        icon="sign out"
                        onClick={this._handleClickLogout}
                        size="mini"
                        floated="right"
                    />
                }
                content="Logout"
            />
        );
    }
}

export default LogoutButtonComponent;
