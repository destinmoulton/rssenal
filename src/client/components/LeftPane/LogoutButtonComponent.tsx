import debug from "debug";
import * as React from "react";

import { Button, Popup } from "semantic-ui-react";

const log = debug("rssenal:LogoutButtonComponent");

export interface ILogoutButtonComponentProps {
    authLogoutUser?: () => void;
}

class LogoutButtonComponent extends React.Component<
    ILogoutButtonComponentProps
> {
    _handleClickLogout = () => {
        log("_handleClickLogout()");
        this.props.authLogoutUser();
    };

    render() {
        log("render()");
        return (
            <Popup
                trigger={
                    <Button
                        className="rss-leftmenu-button-logout"
                        icon="sign out"
                        onClick={this._handleClickLogout}
                        size="mini"
                        floated="right"
                    />
                }
                content="Logout"
            />
        );
    }
}

export default LogoutButtonComponent;
