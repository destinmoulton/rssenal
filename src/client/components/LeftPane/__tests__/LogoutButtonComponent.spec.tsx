import * as React from "react";

import { shallow } from "enzyme";
import LogoutButtonComponent from "../LogoutButtonComponent";

describe("<LogoutButtonComponent />", () => {
    it("renders and matches the snapshot", () => {
        const wrapper = shallow(
            <LogoutButtonComponent logoutUser={jest.fn()} />
        );
        expect(wrapper).toMatchSnapshot();
    });
});
