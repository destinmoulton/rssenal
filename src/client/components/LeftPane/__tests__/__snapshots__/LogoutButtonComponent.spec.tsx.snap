// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoutButtonComponent /> renders and matches the snapshot 1`] = `
<Popup
  content="Logout"
  keepInViewPort={true}
  on="hover"
  position="top left"
  trigger={
    <Button
      as="button"
      className="rss-leftmenu-button-logout"
      floated="right"
      icon="sign out"
      onClick={[Function]}
      role="button"
      size="mini"
    />
  }
/>
`;
